app/models/datawarehouse_node.rb

```diff
@@ -0,0 +1,9 @@
+class DatawarehouseNode < ApplicationRecord
```

Comment:
I'm also not a big fan of having another provider (Datawarehouse) specific model at the root level.

Can we re-use any of these?

```bash
❯ ls app/models/*node*
app/models/compute_node.rb               app/models/container_node.rb
app/models/container_deployment_node.rb  app/models/container_node_performance.rb
```

I guess you are modeling something like a cluster?
Maybe we can even move the above to a common mode?